From cb6dd64dd607c88897b3332e0fdebc24c0a0a590 Mon Sep 17 00:00:00 2001
From: Stewart Smith <stewart@linux.ibm.com>
Date: Tue, 18 Jun 2019 15:48:48 +1000
Subject: [PATCH] struct p9_sbe_msg doesn't need to be packed
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

[ Upstream commit ef691db3533742d9dd6eed1a311472a7c52be94b ]

Only the reg member is sent anywhere (via xscom_write), so the structure
does not need to be packed.

Fixes GCC9 build problem:
hw/sbe-p9.c: In function ‘p9_sbe_msg_send’:
hw/sbe-p9.c:270:9: error: taking address of packed member of ‘struct p9_sbe_msg’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
  270 |  data = &msg->reg[0];
      |         ^~~~~~~~~~~~

Signed-off-by: Stewart Smith <stewart@linux.ibm.com>
---
 include/sbe-p9.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/sbe-p9.h b/include/sbe-p9.h
index 4b839d8ba22b..5ad0422ac91a 100644
--- a/include/sbe-p9.h
+++ b/include/sbe-p9.h
@@ -205,7 +205,7 @@ struct p9_sbe_msg {
 
 	/* Internal queuing */
 	struct list_node	link;
-} __packed;
+};
 
 
 /* Allocate and populate p9_sbe_msg structure */
-- 
2.22.0

